src/main/java/org/apache/commons/imaging/palette/Dithering.java

```diff
@@ -76,42 +72,33 @@ public static void applyFloydSteinbergDithering(final BufferedImage image, final
                 final int nextArgb = palette.getEntry(index);
                 image.setRGB(x, y, nextArgb);
 
-                final int a = (argb >> 24) & 0xff;
-                final int r = (argb >> 16) & 0xff;
-                final int g = (argb >> 8) & 0xff;
-                final int b = argb & 0xff;
+                final int a = (argb >> 24) & MAX_COLOR_VALUE;
+                final int r = (argb >> 16) & MAX_COLOR_VALUE;
+                final int g = (argb >> 8) & MAX_COLOR_VALUE;
+                final int b = argb & MAX_COLOR_VALUE;
 
-                final int na = (nextArgb >> 24) & 0xff;
-                final int nr = (nextArgb >> 16) & 0xff;
-                final int ng = (nextArgb >> 8) & 0xff;
-                final int nb = nextArgb & 0xff;
+                final int na = (nextArgb >> 24) & MAX_COLOR_VALUE;
+                final int nr = (nextArgb >> 16) & MAX_COLOR_VALUE;
+                final int ng = (nextArgb >> 8) & MAX_COLOR_VALUE;
+                final int nb = nextArgb & MAX_COLOR_VALUE;
 
                 final int errA = a - na;
                 final int errR = r - nr;
                 final int errG = g - ng;
                 final int errB = b - nb;
 
-                if (x + 1 < image.getWidth()) {
-                    int update = adjustPixel(image.getRGB(x + 1, y), errA, errR, errG, errB, 7);
-                    image.setRGB(x + 1, y, update);
+                for (int weightIndex = 0; weightIndex < DITHER_WEIGHTS.length; weightIndex++) {
+                    if (x + 1 < image.getWidth() && y + DITHER_WEIGHTS[weightIndex] < image.getHeight()) {
+                        updateImagePixel(image, x + 1, y + DITHER_WEIGHTS[weightIndex], errA, errR, errG, errB, weightIndex);
+                    }
                     if (y + 1 < image.getHeight()) {
-                        update = adjustPixel(image.getRGB(x + 1, y + 1), errA, errR, errG, errB, 1);
-                        image.setRGB(x + 1, y + 1, update);
+                        updateImagePixel(image, x, y + 1, errA, errR, errG, errB, weightIndex);
+                        if (x - 1 >= 0 && y + DITHER_WEIGHTS[weightIndex] < image.getHeight()) {
+                            updateImagePixel(image, x - 1, y + DITHER_WEIGHTS[weightIndex], errA, errR, errG, errB, weightIndex);
+                        }
                     }
                 }
-                if (y + 1 < image.getHeight()) {
-                    int update = adjustPixel(image.getRGB(x, y + 1), errA, errR, errG, errB, 5);
-                    image.setRGB(x, y + 1, update);
-                    if (x - 1 >= 0) {
-                        update = adjustPixel(image.getRGB(x - 1, y + 1), errA, errR, errG, errB, 3);
-                        image.setRGB(x - 1, y + 1, update);
-                    }
-
-                }
             }
         }
     }
-
```

Comment:
-1: You've now made the constructor public, which we do not want since this would allow users to create instances. I added a comment here.

Reply:
@ashishnagpal2498 
It's quite a waste of time to review moving code around, please don't do it.